### PR TITLE
[#37, #67] add missing docs, remove deprecated taxonomy references

### LINKED ISSUE #37: Update fideslang documentation
### Docs Update Description

Now that fideslang lives here full-time, the docs probably need to be revisited and updated to match the true state of fideslang.

### Additional context

There might be some useful docs still in fidesctl for fideslang, but they might be gone by the team this ticket gets worked on so `git history` will come in handy

These changes were also recently made to Fideslang and published as version 1.1.0 - https://ethyca.atlassian.net/wiki/spaces/EN/pages/2345599010/Fideslang+1.1.0+Planning

The ticket for that work is here: https://github.com/ethyca/fideslang/issues/58

### PR DESCRIPTION
**Pull request by conceptualshark**
Closes #37, #67 

### Code Changes

* [x] ported over the remaining fideslang docs from fidesctl with new updates
* [x] Removed all remaining usages of `identifiable`, `nonidentifiable`, `derived`, and `provided` subcategories from:
  * [x] the documentation files
  * [x] the YAML, json, and resource files
* [x] updated the .csvs building the visual taxonomy
* [x] updated the visual taxonomy js to remove deprecated labels
* [x] updated tests to use examples of the new taxonomy

### Steps to Confirm

* [x] docs build succeeds and finds no instances of the deprecated terms
* [x] visual taxonomy is viewable with the new hierarchy 
* [x] tests succeed

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by conceptualshark** [comment]
Decided to break these into two tickets because the taxonomy changes got pretty bloated. Additional updates required by #37 will go into their own PR.

**Comment by conceptualshark** [comment]
@ThomasLaPiana Sorry this is unfortunately large 😬 just grabbing all the remaining taxonomy changes added up

File: mkdocs/docs/resources/dataset.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  ```
+  organization
+    |-> registry (optional)
+        |-> system
+            |-> ** dataset **
+                |-> collections
+                    |-> fields
+  ```
+
+* The schema is represented as a set of "collections" (tables) that contain "fields" (columns).
```
Comment (ThomasLaPiana) [inline mkdocs/docs/resources/dataset.md]: I would make a note here that it can also arbitrarily nested to handle document-type databases (NoSQL, S3, etc.)

**Review by ThomasLaPiana — APPROVED** [review]
Left a single note, but overall this is an amazing PR @conceptualshark 🦾

**Comment by ThomasLaPiana** [comment]
the merge conflict should be pretty simple, we've added the new `is_default` field which is why it is conflicting but the deletion should remain I believe

**Comment by conceptualshark** [comment]
@ThomasLaPiana I believe that should be all set now - thanks for digging through all this! 👍 I think because of the way fideslang is set up, you may need to perform the merge.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,10 @@ The types of changes are:
 * `Security` in case of vulnerabilities.
 
 ## [Unreleased](https://github.com/ethyca/fideslang/compare/1.1.0...main)
+=======
```
Comment (ThomasLaPiana) [inline CHANGELOG.md]: this line is part of the merge conflict

```suggestion
```

**Comment by ThomasLaPiana** [comment]
@conceptualshark oh it wants a review from you but I don't think it accepts a review from the PR author 😆 

Merging!

**Pull request merged**